### DISCUSSION TITLE
wrong arrangement in react-spring

### DISCUSSION THREAD

**Discussion by OmerAdam34** [neuefische/questions#265] [Web] [answered]
Hi, 

I tried to implement a dropdown menu with react spring. But it doesn't work the way i want it to. I tried many things but unfortunately I failed again and again. 

As soon as I click the button, my elements are arranged completely wrong (z-index doesn't work).

the code

`const [show, setShow] = useState(false);

	const contentProps = useSpring({
		opacity: show ? 1 : 0,
		marginTop: show ? 0 : -1500,
	});

	return (
		<div>
			<NavigationButton onClick={() => setShow(a => !a)}>
				<RiMenu5Fill />
			</NavigationButton>

			{show ? (
				<a.div style={contentProps}>
					<List>
						<li>
							<Link href="/">
								<LinkOne>Studio</LinkOne>
							</Link>
						</li>

						<li>
							<Link href="/introduction">
								<LinkOne>Introduction</LinkOne>
							</Link>
						</li>
						<li>
							<Link href="/login">
								<LinkOne>Account</LinkOne>
							</Link>
						</li>
					</List>
				</a.div>
			) : null}
		</div>`

<img width="335" alt="Bildschirmfoto 2022-10-13 um 22 57 52" src="https://user-images.githubusercontent.com/107583280/195709514-7b86a780-53e4-447e-a28e-343bd0845b54.png">

**Comment by pixelass** [comment, marked as answer]
can you send a link to the PR please? 🙏

**Reply by pixelass** [reply]
FYI: you can use `y` instead of `marginTop` to have smoother animations.

(react-spring optimizes it to use `transform: translate3d(...)`)

Take a look at the docs: https://beta.react-spring.dev/docs/getting-started#your-first-animation


Demo: https://codesandbox.io/s/sandpack-project-forked-t1zytx?file=/App.js

**Reply by OmerAdam34** [reply]
Hi Coach G. Sure, here is my pr: https://github.com/OmerAdam34/capstone-project/pull/30#issue-1408916676

**Reply by OmerAdam34** [reply]
> FYI: you can use `y` instead of `marginTop` to have smoother animations.
> 
> (react-spring optimizes it to use `transform: translate3d(...)`)
> 
> Take a look at the docs: https://beta.react-spring.dev/docs/getting-started#your-first-animation
> 
> Demo: https://codesandbox.io/s/sandpack-project-forked-t1zytx?file=/App.js

Ahhh, i´ll try it and give you a feedback :) thanks

**Comment by OmerAdam34** [comment]
This coach G. takes off like a rocket :rocket:
